src/wp-includes/class-wp-navigation-fallback.php

```diff
@@ -0,0 +1,238 @@
+		}
+
+		return null;
+	}
+
+	/**
+	 * Finds the most recently published `wp_navigation` post type.
+	 *
+	 * @since 6.3.0.
+	 * @return WP_Post|null the first non-empty Navigation or null.
+	 */
+	private static function get_most_recently_published_navigation() {
+
+		$parsed_args = array(
+			'post_type'              => 'wp_navigation',
+			'no_found_rows'          => true,
+			'update_post_meta_cache' => false,
+			'update_post_term_cache' => false,
+			'order'                  => 'DESC',
+			'orderby'                => 'date',
+			'post_status'            => 'publish',
+			'posts_per_page'         => 1,
+		);
+
+		$navigation_post = new WP_Query( $parsed_args );
+
+		if ( count( $navigation_post->posts ) > 0 ) {
+			return $navigation_post->posts[0];
+		}
+
+		return null;
+	}
+
+	/**
+	 * Creates a Navigation Menu post from a Classic Menu.
+	 *
+	 * @since 6.3.0.
+	 * @return int|WP_Error The post ID of the default fallback menu or a WP_Error object.
+	 */
+	private static function create_classic_menu_fallback() {
+		// See if we have a classic menu.
+		$classic_nav_menu = static::get_fallback_classic_menu();
+
+		if ( ! $classic_nav_menu ) {
+			return new WP_Error( 'no_classic_menus', __( 'No Classic Menus found.' ) );
+		}
+
+		// If there is a classic menu then convert it to blocks.
+		$classic_nav_menu_blocks = WP_Classic_To_Block_Menu_Converter::convert( $classic_nav_menu );
+
+		if ( empty( $classic_nav_menu_blocks ) ) {
+			return new WP_Error( 'cannot_convert_classic_menu', __( 'Unable to convert Classic Menu to blocks.' ) );
+		}
+
+		// Create a new navigation menu from the classic menu.
+		$classic_menu_fallback = wp_insert_post(
+			array(
+				'post_content' => $classic_nav_menu_blocks,
+				'post_title'   => $classic_nav_menu->name,
+				'post_name'    => $classic_nav_menu->slug,
+				'post_status'  => 'publish',
+				'post_type'    => 'wp_navigation',
+			),
+			true // So that we can check whether the result is an error.
+		);
+
+		return $classic_menu_fallback;
+	}
+
+	/**
+	 * Determine the most appropriate classic navigation menu to use as a fallback.
+	 *
+	 * @since 6.3.0.
+	 * @return WP_Term|null The most appropriate classic navigation menu to use as a fallback.
+	 */
+	private static function get_fallback_classic_menu() {
+		$classic_nav_menus = wp_get_nav_menus();
+
+		if ( ! $classic_nav_menus || is_wp_error( $classic_nav_menus ) ) {
+			return null;
+		}
+
+		$nav_menu = static::get_nav_menu_at_primary_location();
+
+		if ( $nav_menu ) {
+			return $nav_menu;
+		}
+
+		$nav_menu = static::get_nav_menu_with_primary_slug( $classic_nav_menus );
+
+		if ( $nav_menu ) {
+			return $nav_menu;
+		}
+
+		return static::get_most_recently_created_nav_menu( $classic_nav_menus );
+	}
+
+
+	/**
+	 * Sorts the classic menus and returns the most recently created one.
+	 *
+	 * @since 6.3.0.
+	 * @param WP_Term[] $classic_nav_menus Array of classic nav menu term objects.
```

Comment:
We generally separate `since` and `params` with an empty line (see [example](https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#1-functions-class-methods))